### DISCUSSION TITLE
Specify build env location for setuptools.build_meta

### DISCUSSION THREAD

**Discussion by ampx-mg** [pypa/setuptools#3647] [Q&A] [answered]
I am transitioning to pyproject.toml and currently playing around with build backends. Is there a way to specify the location for the build environment? Currently if I run `pip install .`, it gets created in a random `tmp` folder like `/tmp/pip-build-env-5bmqs9ih`, can I somehow change that?

**Comment by webknjaz** [comment, marked as answer]
I don't think so. And also, build backends don't usually change the directory (setuptools is one). It's build frontends, like pip, that create temporary directories and prepare the environment. There's also pypa/build that allows skipping the env preparation but I think it still doesn't let you disable that.

If you want that for whatever reason, you could write your own front-end using the `pep517` library (it is what's used in build and pip). There's also in-tree build frontends but I don't think they get any info about the original source location. Maybe one could smuggle this context via the build config or env vars but that's probably not the best idea.

**Reply by ampx-mg** [reply]
Ahh okay fair enough, thanks for the explanation!
